server/jetstream_api.go

```diff
@@ -27,118 +28,121 @@ import (
+	// Will return JSON response.
+	JSApiConsumers  = "$JS.API.CONSUMERS.*"
+	JSApiConsumersT = "$JS.API.CONSUMERS.%s"
 
-	// JetStreamConsumerInfo is for obtaining general information about a consumer.
+	// JSApiConsumerInfo is for obtaining general information about a consumer.
 	// Will return JSON response.
-	JetStreamConsumerInfo  = "$JS.STREAM.*.CONSUMER.*.INFO"
-	JetStreamConsumerInfoT = "$JS.STREAM.%s.CONSUMER.%s.INFO"
+	JSApiConsumerInfo  = "$JS.API.CONSUMER.INFO.*.*"
+	JSApiConsumerInfoT = "$JS.API.CONSUMER.INFO.%s.%s"
 
-	// JetStreamDeleteConsumer is the endpoint to delete consumers.
-	// Will return +OK on success and -ERR on failure.
-	JetStreamDeleteConsumer  = "$JS.STREAM.*.CONSUMER.*.DELETE"
-	JetStreamDeleteConsumerT = "$JS.STREAM.%s.CONSUMER.%s.DELETE"
+	// JSApiDeleteConsumer is the endpoint to delete consumers.
+	// Will return JSON response.
+	JSApiConsumerDelete  = "$JS.API.CONSUMER.DELETE.*.*"
+	JSApiConsumerDeleteT = "$JS.API.CONSUMER.DELETE.%s.%s"
 
-	// JetStreamAckT is the template for the ack message stream coming back from an consumer
-	// when they ACK/NAK, etc a message.
-	JetStreamAckT   = "$JS.ACK.%s.%s"
-	jetStreamAckPre = "$JS.ACK."
+	// JSApiRequestNextT is the prefix for the request next message(s) for a consumer in worker/pull mode.
+	JSApiRequestNextT = "$JS.API.CONSUMER.MSG.NEXT.%s.%s"
 
-	// JetStreamRequestNextT is the prefix for the request next message(s) for a consumer in worker/pull mode.
-	JetStreamRequestNextT = "$JS.STREAM.%s.CONSUMER.%s.NEXT"
+	///////////////////////
+	// FIXME(dlc)
+	///////////////////////
 
-	// JetStreamMsgBySeqT is the template for direct requests for a message by its stream sequence number.
-	JetStreamMsgBySeqT = "$JS.STREAM.%s.MSG.BYSEQ"
+	// JetStreamAckT is the template for the ack message stream coming back from a consumer
+	// when they ACK/NAK, etc a message.
+	// FIXME(dlc) - What do we really need here??
+	jsAckT   = "$JS.ACK.%s.%s"
+	jsAckPre = "$JS.ACK."
 
-	// JetStreamAdvisoryPrefix is a prefix for all JetStream advisories.
-	JetStreamAdvisoryPrefix = "$JS.EVENT.ADVISORY"
+	// JSAdvisoryPrefix is a prefix for all JetStream advisories.
+	JSAdvisoryPrefix = "$JS.EVENT.ADVISORY"
 
-	// JetStreamMetricPrefix is a prefix for all JetStream metrics.
-	JetStreamMetricPrefix = "$JS.EVENT.METRIC"
+	// JSMetricPrefix is a prefix for all JetStream metrics.
+	JSMetricPrefix = "$JS.EVENT.METRIC"
 
-	// JetStreamMetricConsumerAckPre is a metric containing ack latency.
-	JetStreamMetricConsumerAckPre = JetStreamMetricPrefix + ".CONSUMER_ACK"
+	// JSMetricConsumerAckPre is a metric containing ack latency.
+	JSMetricConsumerAckPre = "$JS.EVENT.METRIC.CONSUMER.ACK"
 
 	// JetStreamAdvisoryConsumerMaxDeliveryExceedPre is a notification published when a message exceeds its delivery threshold.
-	JetStreamAdvisoryConsumerMaxDeliveryExceedPre = JetStreamAdvisoryPrefix + ".MAX_DELIVERIES"
+	JSAdvisoryConsumerMaxDeliveryExceedPre = "$JS.EVENT.ADVISORY.CONSUMER.MAX_DELIVERIES"
```

Comment:
this will limit us in the future, to subscribe to all advisories related to a stream we have to do

`$JS.EVENT.ADVISORY.*.*.<STREAM>.*` but that only works if we never add advisories that doesnt relate to a stream - `CONSUMER` here - we might want advisories for JS internals which means this will then not work.

Ditto with metrics - metrics especially we'll want non stream/consumer stuff but I'll need to be able to create subjects that consumes all for a stream or all for a consumer

Reply:
Not sure I follow, but on the wire having the identifiers at the end was much cleaner IMO then randomly in the subject. Happy to discuss more.

Reply:
Discussed, will leave for now.